src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[INITIAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndex(index, size + 1);
+        ensureSize(size + 1);
+        if (index < size) {
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+        }
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null || list.size() == 0) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        elements[index] = value;
     }
 
     @Override
+    @SuppressWarnings("unchecked")
```

Comment:
You should not use this annotation
```suggestion
```

Reply:
Do you mean there should not be code pieces that cause unchecked warnings?

Reply:
There are still unchecked warnings by the compiler.
Do you suppose it is better to keep it?